packages/@react-aria/dnd/src/useDropIndicator.ts

```diff
@@ -61,9 +61,16 @@ export function useDropIndicator(props: DropIndicatorProps, state: DroppableColl
       itemText: getText(target.key)
     });
   } else {
-    let before = target.dropPosition === 'before'
-      ? collection.getKeyBefore(target.key)
-      : target.key;
+    let before = null;
+    if (target.dropPosition === 'before') {
+      let keyBefore = collection.getKeyBefore(target.key);
+      // TODO: move logic to TableView
+      if (keyBefore && !keyBefore.toString().startsWith('headerrow-')) {
+        before = keyBefore;
+      }
+    } else {
+      before = target.key;
+    }
```

Comment:
This handles the case where it was announcing `"Insert between null and Item One"` instead of `"Insert before Item One"`. It was detecting the header row as the key before. Should probably handle this somewhere else if possible though.

Reply:
Thinking out loud, but perhaps we could add and additional check to the `before` and `after` logic here where we check the collection's first and last key via `getFirstKey` and `getLastKey`? If the first key matches the `target.key` then we can set `before` to null and do something similar for last key to guard against the case where we have nodes after what we consider is the "last" row.

The problem is that this relies on the collection's `getFirstKey` and `getLastKey` returning the row keys, which breaks in the case where the Table collection's bodyNodes are Sections instead... Perhaps `useDropIndicator` should accept some user defined options for determining `keyBefore` and `keyAfter` so we can customize this logic on a case by case basis? That would allow us to omit certain node types that we know don't make sense in a DnD context like the header row

Reply:
This seems like a good solution so I'll implement this for now!